ebean-api/src/main/java/io/ebean/Database.java

```diff
@@ -866,7 +871,7 @@ public interface Database {
    * the bean does not exist then this returns false indicating that nothing was
    * deleted. However, if JDBC batch mode is used then this always returns true.
    */
-  boolean delete(Object bean, Transaction transaction) throws OptimisticLockException;
+  boolean delete(Object bean, @Nullable Transaction transaction) throws OptimisticLockException;
```

Comment:
Really? (haven't checked it). But if JavaDoc states `... with an explicit transaction.` on an API method it feels weird to me that this explicit transaction can actually be `null`. Could very well hide some coding problems in user code.

If that is really true, then JavaDoc should describe that you can pass in `null` as a transaction and what this means if you do so. Same for all the other methods taking an explicit transaction as input parameter. In that case JavaDoc of the equivalent method without transaction parameter could say something like `this method behaves exactly as passing a null transaction to delete(bean, transaction)` and link to the other method. Would avoid duplicating JavaDoc on all these methods.

Reply:
> Really? (haven't checked it). But if JavaDoc states ... with an explicit transaction.

Yes, the implementation checks it, and the methods without transaction parameter just pass null there.

> If that is really true, then JavaDoc should describe that you can pass in null as a transaction and what this means if you do so. 

ACK